diff --git a/src/oca/ruby/opennebula/image.rb b/src/oca/ruby/opennebula/image.rb
index cc89dc1..061d800 100644
--- a/src/oca/ruby/opennebula/image.rb
+++ b/src/oca/ruby/opennebula/image.rb
@@ -64,7 +64,7 @@ module OpenNebula
             "CONTEXT"   => "CX"
         }
 
-        DISK_TYPES=%w{FILE CD_ROM BLOCK RBD}
+        DISK_TYPES=%w{FILE CD_ROM BLOCK RBD SHEEPDOG}
 
         # Creates an Image description with just its identifier
         # this method should be used to create plain Image objects.
